### ISSUE TITLE
emedded pandoc 3.1.1 is insecure 

### ISSUE THREAD

**Issue by jonassmedegaard** [quarto-dev/quarto-cli#6346]
### Bug description

Embedded pandoc 3.1.1 is vulnerable to CVE-2023-35936.

Please issue a new release which embeds pandoc 3.1.6 or newer.

**Comment by jonassmedegaard** [comment]
Development releases exist that embeds adequately new Pandoc.
Updated to emphasize that I mean to request a new *stable* release.

**Comment by cscheid** [comment]
From the CVE:

> Pandoc is susceptible to an arbitrary file write vulnerability, which can be triggered by providing a specially crafted image element in the input when generating files using the `--extract-media` option or outputting to PDF format.

Do you have a downstream scanning tool that is asking you to do upgrade? I'm asking to understand the issue better. Specifically, the threat model here involves the creation of a malicious file. However, quarto (by design) does not protect against this particular threat, since we allow (by design) execution of arbitrary code to begin with.

**Comment by jonassmedegaard** [comment]
My "scanning tool" is that a) I am [the Debian package maintainer for pandoc](https://tracker.debian.org/pkg/pandoc), and b) one of my fellow Debian developers is Guilhem Moulin who found and reported a followup for this CVE and notified me about it. :-)

I can understand if quarto by design allows execution of arbitrary code **which is passed to it from the user** - pandoc supports that as well.  But are you saying that quarto don't care about code getting executed **when the user does not ask for execution of code** as well?

**Comment by cscheid** [comment]
> But are you saying that quarto don't care about code getting executed when the user does not ask for execution of code as well?

Of course I'm not saying that. I'm asking about your context so we can prioritize the needed fixes on a finite time budget.

**Comment by jonassmedegaard** [comment]
Sorry, I was not trying to twist your words - I was sincerely trying to verify what in my impression you seemed to imply. My apology for misunderstanding you.

(I still don't really understand what you were getting at, and am happy to answer any further questions you might have)

**Comment by cscheid** [comment]
No apologies necessary at all! Just miscommunication, don't worry about it.

We get upstream CVE scanning requests from customers which mean that they are _completely blocked_. That raises our internal priority to "fix it absolutely right now".

I'm asking about context because "when the user does not ask for execution of code as well?" is fundamentally a confused deputy issue. In that sense, "the user not asking for execution of code" means malicious input, and as far as the threat is concerned, we're already vulnerable to malicious input, and closing the pandoc hole doesn't close the "the user not asking for execution of code".

It _does_ close a specific instance of "it's hard for user to see that this particular code is malicious" threat specific to this CVE. And we want to close this bug very soon. But porting 3.1.6 back to quarto 1.3 is not as straightforward as it seems because of the other Pandoc changes that we need to backport, and it's a good bit of work on our side.

tl;dr: I was just trying to understand if this meant "stop the world and fix this literally now" or "fix it in the next few days" category.

**Comment by jonassmedegaard** [comment]
Ah, thanks for the clarification.

This is - to me, at least - in the "fix in the next few days (or whenever you are back from your nice holiday, or have pleasantly entered into it, or whatever)" category.

**Comment by jonassmedegaard** [comment]
If you need help backporting the fix then please shout - or perhaps take inspiration from the 02023* patches [here](https://salsa.debian.org/haskell-team/pandoc/-/tree/debian/latest/debian/patches) for my backport to pandoc 2.17.1.1.

**Comment by cscheid** [comment]
> If you need help backporting the fix then please shout 

I appreciate the offer!, but the backporting necessary involves changes in Pandoc readers/writers. Unfortunately, I don't think you'd be able to help fixing our internal Lua filters :)

**Comment by cderv** [comment]
> If you need help backporting the fix then please shout - or perhaps take inspiration from the 02023* patches [here](https://salsa.debian.org/haskell-team/pandoc/-/tree/debian/latest/debian/patches) for my backport to pandoc 2.17.1.1.

@jonassmedegaard as I see you patch know that we are not building Pandoc. We are using Pandoc release directly. 

Our 1.4 pre-release build already use 3.1.6 by the way (and the 3.1.4 is the pandoc version patched IIUC). 1.4 will be the next stable release at fall probably (no commitment though on the date). 

Just adding those notes for context in case it was not clear what we do exactly 😃

**Comment by cderv** [comment]
> Just wanted to make the security concern explicit for you guys, in case you had different priorities.

And thanks a lot for that! We really appreciate it !
